proposals/lip-0041.md

```diff
@@ -97,23 +111,300 @@ In this section, we describe the properties of an auth account and specify their
+```python
+def execute(trs: Transaction) -> None:
+    # Verify the signatures 
+    message = encode(
+        schema = multisigRegMsgSchema,
+        object = {
+            "address": senderAddress,
+            "nonce": trs.nonce,
+            "numberOfSignatures": trs.params.numberOfSignatures,
+            "mandatoryKeys": trs.params.mandatoryKeys,
+            "optionalKeys": trs.params.optionalKeys 
+        }
+    )
+    allKeys = trs.params.mandatoryKeys + trs.params.optionalKeys
+    for signature, key in zip(trs.params.signatures, allKeys):
+        if not verifyMessageSig(key, MESSAGE_TAG_MULTISIG_REG, networkIdentifier, message, signature):
+            emitPersistentEvent(
+                moduleID = MODULE_ID_AUTH,
+                typeID = TYPE_ID_MULTISIGNATURE_GROUP_REGISTERED,
+                data = encode(
+                    schema = invalidSigDataSchema,
+                    object = {
+                        "numberOfSignatures": trs.params.numberOfSignatures,
+                        "mandatoryKeys": trs.params.mandatoryKeys,
+                        "optionalKeys": trs.params.optionalKeys,
+                        "failingPublicKey": key,
+                        "failingSignature": signature
+                    }
+                ),
+                topics=[senderAddress]
+            )
+            raise Exception(f'Invalid signature for public key {key.hex()}.')
+
+    senderAddress = 20-byte address derived from trs.senderPublicKey
+    authAccount(senderAddress).numberOfSignatures = trs.params.numberOfSignatures
+    authAccount(senderAddress).mandatoryKeys = trs.params.mandatoryKeys
+    authAccount(senderAddress).optionalKeys = trs.params.optionalKeys
+    
+    emitEvent(
+        moduleID = MODULE_ID_AUTH,
+        typeID = TYPE_ID_MULTISIGNATURE_GROUP_REGISTERED,
+        data = encode(
+            schema = multisigRegDataSchema,
+            object = {
+                "numberOfSignatures": trs.params.numberOfSignatures,
+                "mandatoryKeys": trs.params.mandatoryKeys,
+                "optionalKeys": trs.params.optionalKeys
+            }
+        ),
+        topics=[senderAddress]
+    )
+```
+
+The function `verifyMessageSig` is defined in [LIP 0062](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0062.md#specification) and `networkIdentifier` is the network identifier of the chain.
+
+### Internal Functions
+
+The Auth module has the following internal function.
+
+#### isMultisignatureAccount
+
+This function checks whether the account corresponding to the input address has registered a multisignature group.
+
+```python
+def isMultisignatureAccount(address: Address) -> bool:
+    if there is no entry in the in the auth data substore with storeKey == address:
+        return False
+
+    if len(authAccount(senderAddress).mandatoryKeys) == len(authAccount(senderAddress).mandatoryKeys) == 0:
```

Comment:
I don't understand this condition with two `==`. Shouldn't it be sufficient to check 
```
authAccount(senderAddress).numberOfSignatures == 0
```
Of course, we can also check that all 3 properties in the Auth store have the default values.

Reply:
True, maybe I was overthinking this